services/blockchain-connector/tests/functional/decoder.test.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const { ServiceBroker } = require('moleculer');
+
+const {
+	block,
+	blockWithTransaction,
+	encodedBlock,
+	invalidEncodedBlock,
+	invalidEncodedTransaction,
+	encodedBlockWithTransaction,
+} = require('../constants/blocks');
+
+const {
+	transaction,
+	encodedTransaction,
+} = require('../constants/transactions');
+const config = require('../../config');
+
+const broker = new ServiceBroker({
+	transporter: config.transporter,
+	logLevel: 'warn',
+	requestTimeout: 15 * 1000,
+	logger: console,
+});
+
+describe('Functional tests for decoder', () => {
+	beforeAll(() => broker.start());
+	afterAll(() => broker.stop());
+
+	it('decode Transaction', async () => {
+		const result = await broker.call('connector.decodeTransactionSerialized', { encodedTransaction });
+		transaction.size = result.size;
+		expect(Object.keys(result)).toEqual(Object.keys(transaction));
+		expect(result).toMatchObject(transaction);
+	});
+
+	it('decode block with transaction', async () => {
+		const result = await broker.call('connector.decodeBlockSerialized', { encodedBlock: encodedBlockWithTransaction });
+		expect(result).toMatchObject({
+			header: expect.any(Object),
+			assets: expect.any(Object),
+			transactions: expect.any(Object),
+		});
+		blockWithTransaction.transactions[0].size = result.transactions[0].size;
```

Comment:
Please remove this line and ensure the constants already have the expected values.